packages/run-protocol/src/reserve/collateralReserve.js

```diff
@@ -0,0 +1,218 @@
+  const ammPublicFacet = E(zcf.getZoeService()).getPublicFacet(
+    ammInstance.value,
+  );
+
+  const addIssuer = async (issuer, keyword) => {
+    const brand = await E(issuer).getBrand();
+
+    const liquidityIssuer = E(ammPublicFacet).getLiquidityIssuer(brand);
+    const liquidityBrand = await E(liquidityIssuer).getBrand();
+
+    brandForKeyword.init(keyword, [brand, liquidityBrand]);
+    keywordForBrand.init(brand, keyword);
+    return Promise.all([
+      zcf.saveIssuer(issuer, keyword),
+      zcf.saveIssuer(liquidityIssuer, makeLiquidityKeyword(keyword)),
+    ]);
+  };
+
+  const getKeywordsForBrand = brand => {
+    assert(
+      keywordForBrand.has(brand),
+      `Please call addIssuer for brand ${brand}`,
+    );
+    return keywordForBrand.get(brand);
+  };
+
+  const { initialPoserInvitation, feeMintAccess } = privateArgs;
+
+  /** a powerful object; can modify the invitation */
+  const paramManager = await makeParamManager(
+    zcf.getZoeService(),
+    ammInstance.value,
+    initialPoserInvitation,
+  );
+
+  const { wrapPublicFacet, wrapCreatorFacet, getElectorate } =
+    handleParamGovernance(zcf, paramManager);
+
+  const electorateInvAmt = getElectorate();
+  // compare invitation amount from privateArgs with electorateParam from terms
+  assert(
+    keyEQ(electorateInvAmt, electorateParam.value),
+    X`electorate amount (${electorateParam.value} didn't match ${electorateInvAmt}`,
+  );
+
+  // const brandToKeyword = makeStore('keyword');
+  // should liquidiity tokens go in a separate seat?
+  const { zcfSeat: collateralSeat } = zcf.makeEmptySeatKit();
+  const getAllocations = () => {
+    return collateralSeat.getCurrentAllocation();
+  };
+
+  // Anyone can deposit collateral to the reserve.
+  const addCollateralHook = async seat => {
+    const {
+      give: { Collateral: amountIn },
+    } = seat.getProposal();
+
+    const keyword = getKeywordsForBrand(amountIn.brand);
+    seat.decrementBy(harden({ Collateral: amountIn }));
+    collateralSeat.incrementBy(harden({ [keyword]: amountIn }));
+
+    zcf.reallocate(collateralSeat, seat);
+    seat.exit();
+
+    return 'added Collateral to the Reserve';
+  };
+
+  const makeAddCollateralInvitation = () =>
+    zcf.makeInvitation(addCollateralHook, 'Add Collateal');
+
+  /** @type {ZCFMint} */
+  const runMint = await zcf.registerFeeMint('RUN', feeMintAccess);
+
+  const addLiquidityToAmmPool = async (collateralAmount, runAmount) => {
+    // verify we have the funds
+    const keyword = getKeywordsForBrand(collateralAmount.brand);
+    if (
+      !AmountMath.isGTE(
+        collateralSeat.getCurrentAllocation()[keyword],
+        collateralAmount,
+      )
+    ) {
+      throw new Error('insufficient reserves for that transaction');
+    }
+
+    // create the RUN
+    const offerToSeat = runMint.mintGains(harden({ RUN: runAmount }));
+    offerToSeat.incrementBy(
+      collateralSeat.decrementBy(
+        harden({
+          [keyword]: collateralAmount,
```

Comment:
to disambiguate in the context consider
```suggestion
          [collateralBrand]: collateralAmount,
```

Reply:
I changed it to `[collateralKeyword]`.